### PR TITLE
error-pages repository and configuration

### PR DESCRIPTION
**Pull request by giorgiosironi**

### CODE REVIEW

**Comment by thewilkybarkid** [comment]
Should do the same for the default server.

File: salt/journal/config/etc-nginx-sites-enabled-journal.conf [hunk 1]
```diff
@@ -57,4 +57,14 @@ server {
 
     access_log /var/log/nginx/journal.access.log combined_with_time;
     error_log /var/log/nginx/journal.error.log notice;
+
+    error_page 404 /error/404.html;
+    error_page 410 /error/410.html;
+    error_page 400 401 402 403 405 406 407 408 409 411 412 413 414 415 416 417 418 421 422 423 424 426 428 429 431 444 451 495 496 497 /error/4xx.html;
```
Comment (thewilkybarkid) [inline salt/journal/config/etc-nginx-sites-enabled-journal.conf]: I don't think there's a better way to do this (ie a wildcard). I took the list of codes from Wikipedia with the Nginx extras, but had to remove `499 Client Closed Request` as it doesn't allow it.

**Comment by giorgiosironi** [comment]
Now has no effect by default, but after killing php-fpm:
```
[13:31:09][elife@journal--vagrant:~]$ curl -v localhost/
* Hostname was NOT found in DNS cache
*   Trying 127.0.0.1...
* Connected to localhost (127.0.0.1) port 80 (#0)
> GET / HTTP/1.1
> User-Agent: curl/7.35.0
> Host: localhost
> Accept: */*
> 
< HTTP/1.1 502 Bad Gateway
* Server nginx/1.4.6 (Ubuntu) is not blacklisted
< Server: nginx/1.4.6 (Ubuntu)
< Date: Thu, 12 Jan 2017 13:31:23 GMT
< Content-Type: text/html
< Content-Length: 242
< Connection: keep-alive
< ETag: "58776461-f2"
< 
<!doctype html>
<html lang="en">
<head>

    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1,shrink-to-fit=no">

    <title>Error | eLife</title>
</head>
<body>
Server error

</body>
</html>
```

File: salt/journal/config/etc-nginx-sites-enabled-journal.conf (hunk 1 above)
Comment (giorgiosironi) [inline salt/journal/config/etc-nginx-sites-enabled-journal.conf]: No wildcard that I know of (it would also probably catch 301 and similar)

**Comment by thewilkybarkid** [comment]
Seems like `fastcgi_intercept_errors off;` is needed at the `http` level to get that to work.

**Pull request merged**